app/assets/javascripts/discourse/components/choose-message.js.es6

```diff
@@ -0,0 +1,65 @@
+import debounce from "discourse/lib/debounce";
+import { searchForTerm } from "discourse/lib/search";
+import { observes } from "ember-addons/ember-computed-decorators";
+
+export default Ember.Component.extend({
+  loading: null,
+  noResults: null,
+  messages: null,
+
+  @observes("messageTitle")
+  messageTitleChanged() {
+    this.setProperties({
+      loading: true,
+      noResults: true,
+      selectedTopicId: null
+    });
+    this.search(this.get("messageTitle"));
+  },
+
+  @observes("messages")
+  messagesChanged() {
+    const messages = this.get("messages");
+    if (messages) {
+      this.set("noResults", messages.length === 0);
+    }
+    this.set("loading", false);
+  },
+
+  search: debounce(function(title) {
+    const self = this,
```

Comment:
I would like us to stop using `self = this` and instead use ES6 arrow functions for keeping scope.